core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/internal/DefaultHttpSignRequest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.auth.spi.HttpSignRequest;
+import software.amazon.awssdk.http.auth.spi.SignerProperty;
+import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
+
+@SdkInternalApi
+// public because it is in different (sub) package
+public final class DefaultHttpSignRequest<PayloadT> implements HttpSignRequest<PayloadT> {
+
+    private final Class<PayloadT> payloadType;
+    private final SdkHttpRequest request;
+    private final Optional<PayloadT> payload;
+    private final HashMap<SignerProperty<?>, Object> properties;
```

Comment:
`Map<SignerProperty<?>, Object>`?

Reply:
Oh 🤦 I missed that you were talking about Map instead of HashMap and thought you were talking about using `?` instead of `Object`. Of course we should use Map, my eyes didn't even see the `Hash`!